packages/components/addon/modifiers/hds-clipboard.js

```diff
@@ -0,0 +1,177 @@
+      // selection.removeAllRanges();
+      // range.selectNodeContents(targetElement);
+      // selection.addRange(range);
+      // textToCopy = selection.toString();
+      // selection.removeAllRanges();
+    }
+  }
+  return textToCopy;
+};
+
+export const writeTextToClipboard = async (textToCopy) => {
+  // finally copy the text to the clipboard using the Clipboard API
+  // https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API
+  if (textToCopy) {
+    try {
+      // notice: the "clipboard-write" permission is granted automatically to pages when they are in the active tab
+      // https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/write
+      await navigator.clipboard.writeText(textToCopy);
+      // DEBUG uncomment this for easy debugging
+      // console.log('success', textToCopy);
+      return true;
+    } catch (error) {
+      // clipboard write failed
+      console.log(
+        '`hds-clipboard` modifier - an error occurred while writing to the clipboard - check your browser permissions',
+        textToCopy,
+        error
+      );
+      return false;
+    }
+  } else {
+    return false;
+  }
+};
+
+export const copyToClipboard = async (text, target) => {
+  let textToCopy;
+
+  if (text) {
+    textToCopy = getTextToCopy(text);
+  } else if (target) {
+    const targetElement = getTargetElement(target);
+    textToCopy = getTextToCopyFromTargetElement(targetElement);
+  } else {
+    assert(
+      '`hds-clipboard` modifier - either a `text` or a `target` argument is required'
+    );
+  }
+  const success = await writeTextToClipboard(textToCopy);
+  return success;
+};
+
+export default class HdsClipboardModifier extends Modifier {
+  didSetup = false;
+
+  modify(element, positional, named) {
+    assert(
+      '`hds-clipboard` modifier - the modifier must be applied to an element',
+      element
+    );
+
+    const { text, target, onSuccess, onError } = named;
+    this.text = text;
+    this.target = target;
+    this.onSuccess = onSuccess;
+    this.onError = onError;
+
+    if (!this.didSetup) {
+      this.#listenClick(element);
+      this.didSetup = true;
+    }
+  }
+
+  // add a "click" event listener to the element
+  #listenClick(element) {
+    this.listener = element.addEventListener('click', (e) => this.#onClick(e));
+  }
+
+  // defines a new `copyToClipboard` action on each click event
+  async #onClick(e) {
+    const trigger = e.delegateTarget || e.currentTarget;
+    const success = await copyToClipboard(this.text, this.target);
+
+    // fire the `onSuccess/onError` callbacks (if provided)
+    const args = {
+      triggger: trigger,
+      text: this.text,
+      target: this.target,
+    };
+    if (success) {
+      if (typeof this.onSuccess === 'function') {
+        this.onSuccess(args);
+      }
+    } else {
+      if (typeof this.onError === 'function') {
+        this.onError(args);
+      }
+    }
+  }
+
+  willDestroy() {
+    this.listener.destroy();
```

Comment:
This won't do anything because `this.listener` will never be defined since  `[addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#return_value)` returns `undefined`. You'd need to call `removeEventListener` with the same handler reference to do the clean up.